File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> list = new ArrayList<>();
        helper(root, k, list);
        return list.get(k - 1);
    }
    private void helper(TreeNode root, int k, List list) {
        if (root == null) {
            return;
        }
        helper(root.left, k, list);
        list.add(root.val);
        if (list.size() == k) {
            return;
        }
        helper(root.right, k, list);
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def helper(root, greatest):
            if not root:
                return 0
            new_greatest = max(greatest, root.val)
            if root.val >= greatest:
                return 1 + helper(root.left, new_greatest) + helper(root.right, new_greatest)
            return helper(root.left, new_greatest) + helper(root.right, new_greatest)
        return helper(root, -float('inf'))